client/src/js/samples/components/__tests__/Rights.test.js

```diff
@@ -0,0 +1,168 @@
+        shallow(<SampleRights {...props} />);
+        expect(props.onListGroups).toHaveBeenCalled();
+    });
+
+    it("should call handleChangeGroup() when input is changed", () => {
+        const wrapper = shallow(<SampleRights {...props} />);
+        wrapper
+            .find("Input")
+            .at(0)
+            .simulate("change", e);
+        expect(props.onChangeGroup).toHaveBeenCalledWith("foo", "bar");
+    });
+
+    it("should call handleChangeRights() when input is changed", () => {
+        const wrapper = shallow(<SampleRights {...props} />);
+        wrapper
+            .find("Input")
+            .at(1)
+            .simulate("change", e);
+        expect(props.onChangeRights).toHaveBeenCalledWith("foo", "group", "bar");
+    });
+
+    it("should return LoadingPlaceholder when[this.props.groups=null]", () => {
+        props.groupd = null;
+        const wrapper = shallow(<SampleRights {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it("should return Not allowed panel when[this.props.canModifyRights=false]", () => {
+        props.canModifyRights = false;
+        const wrapper = shallow(<SampleRights {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+});
+
+describe("mapStateToProps()", () => {
+    const state = {
+        samples: {
+            detail: {
+                all_read: true,
+                all_write: false,
+                group: "baz",
+                group_read: true,
+                group_write: false,
+                all_read: true,
+                user: { id: "Baz" },
+                id: "Boo"
+            }
+        },
+        account: { id: "foo", administrator: true },
+        groups: { documents: "bar" }
+    };
+
+    it("should return props when getCanModifyRights returns false", () => {
+        getCanModifyRights.mockReturnValue(false);
+
+        const props = mapStateToProps(state);
+        expect(props).toEqual({
+            accountId: "foo",
+            all_read: true,
+            all_write: false,
+            canModifyRights: false,
+            group: "baz",
+            group_read: true,
+            group_write: false,
+            groups: "bar",
+            isAdmin: true,
+            ownerId: "Baz",
+            sampleId: "Boo"
+        });
+    });
+
+    it("should return props when getCanModifyRights returns true", () => {
+        getCanModifyRights.mockReturnValue(true);
+
+        const props = mapStateToProps(state);
+        expect(props).toEqual({
+            accountId: "foo",
+            all_read: true,
+            all_write: false,
+            canModifyRights: true,
+            group: "baz",
+            group_read: true,
+            group_write: false,
+            groups: "bar",
+            isAdmin: true,
+            ownerId: "Baz",
+            sampleId: "Boo"
+        });
+    });
+});
+
+describe("mapDispatchToProps()", () => {
+    let state = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['state' is never reassigned. Use 'const' instead.](https://app.codacy.com/manual/virtool/virtool/pullRequest?prid=4748572)